aws/resource_aws_acm_certificate.go

```diff
@@ -283,7 +281,9 @@ func resourceAwsAcmCertificateRead(d *schema.ResourceData, meta interface{}) err
 			return resource.NonRetryableError(fmt.Errorf("Error describing certificate: %s", err))
 		}
 
-		d.Set("domain_name", resp.Certificate.DomainName)
+		// To be consistent with other AWS services that do not accept a trailing period,
+		// we remove the suffix from the Fully Qualified Domain Name of the Certificate returned from the API
+		d.Set("domain_name", trimTrailingPeriod(aws.StringValue(resp.Certificate.DomainName)))
```

Comment:
Since the ACM service does not accept nor return the trailing period, this should be extraneous and the original logic can be left.

```suggestion
		d.Set("domain_name", resp.Certificate.DomainName)
```